app/src/main/java/de/stephanlindauer/criticalmaps/adapter/ChatMessageAdapter.java

```diff
@@ -12,65 +12,77 @@
 import java.util.Locale;
 import java.util.TimeZone;
 
+import butterknife.Bind;
+import butterknife.ButterKnife;
 import de.stephanlindauer.criticalmaps.R;
-import de.stephanlindauer.criticalmaps.utils.TimeToWordStringConverter;
 import de.stephanlindauer.criticalmaps.interfaces.IChatMessage;
-import de.stephanlindauer.criticalmaps.vo.chat.OutgoingChatMessage;
+import de.stephanlindauer.criticalmaps.utils.TimeToWordStringConverter;
 import de.stephanlindauer.criticalmaps.vo.chat.ReceivedChatMessage;
 
-public class ChatMessageAdapter extends ArrayAdapter<IChatMessage> {
-
-    private ArrayList<IChatMessage> chatMessages;
-    private Context context;
+public class ChatMessageAdapter extends RecyclerView.Adapter<ChatMessageAdapter.OutgoingChatMessageViewHolder> {
 
+    private final ArrayList<IChatMessage> chatMessages;
 
-    public ChatMessageAdapter(Context context, int layoutResourceId, ArrayList<IChatMessage> chatMessages) {
-        super(context, layoutResourceId, chatMessages);
+    public ChatMessageAdapter(ArrayList<IChatMessage> chatMessages) {
         this.chatMessages = chatMessages;
-        this.context = context;
     }
 
-    @Override
-    public View getView(int position, View convertView, ViewGroup parent) {
-        LayoutInflater inflater = (LayoutInflater) context
-                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+    public class OutgoingChatMessageViewHolder extends RecyclerView.ViewHolder {
 
-        IChatMessage currentMessage = chatMessages.get(position);
+        @Bind(R.id.firstLine)
+        TextView labelView;
 
-        if (currentMessage instanceof ReceivedChatMessage)
-            return buildReceivedMessageView((ReceivedChatMessage) currentMessage, inflater, parent);
-        else if (currentMessage instanceof OutgoingChatMessage)
-            return buildOutgoingMessageView((OutgoingChatMessage) currentMessage, inflater, parent);
+        @Bind(R.id.secondLine)
+        TextView valueView;
 
-        return null;
-    }
+        private final DateFormat dateFormatter = DateFormat.getDateTimeInstance(DateFormat.DEFAULT, DateFormat.SHORT, Locale.getDefault());
```

Comment:
@ligi I wonder if this can be put as a `static` field somewhere to being shared among the `ViewHolder` instances to lower their footprint. What's you opinion on this?